### PR TITLE
REF: de-duplicate NDFrame.take, remove Manager.take keyword

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3916,8 +3903,14 @@ def _take(
                 and is_range_indexer(indices, len(self))
             ):
                 return self.copy(deep=None)
+        elif self.ndim == 1:
+            # TODO: be consistent here for DataFrame vs Series
```
Comment (jbrockmendel) [inline pandas/core/generic.py:3907]: @phofl following this ill do do the deprecation we discussed

Comment (phofl) [reply pandas/core/generic.py:3907]: Sounds good!

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @jbrockmendel

**Comment by phofl** [comment]
@jbrockmendel Looks like this could have caused a performance regression. https://asv-runner.github.io/asv-collection/pandas/#indexing.NumericSeriesIndexing.time_iloc_array?p-index_structure='unique_monotonic_inc'&p-dtype=%3Cclass%20'numpy.int64'%3E

Could not reproduce locally though, do you have an idea?

**Comment by jbrockmendel** [comment]
It's plausible.  This PR makes us go through Manager.take whereas the old version operated directly on ._values/.index.  In retrospect its not that surprising, though I'd have hoped for the effect to be smaller.

Restoring Series.take (the rest of this should be OK) should fix it.

If we go for something like ExtensionManager mentioned in #51471 then we'd need to have basically everything go through a Manager method, in which case we'd need to optimize Manager.take somehow.

**Comment by phofl** [comment]
Yep adding Series.take back would get performance back, but as you said not really a long term solution unfortunately

**Pull request merged**